src/preprocess/query_generator.py

```diff
@@ -0,0 +1,225 @@
+          "inner join programstage ps on ps.programstageid=psi.programstageid "
+          "inner join program p on p.programid=ps.programid "
+          "where p.uid in " + trackers + ");" + "\n")
+    f.write(
+        "delete from programstageinstancecomments where programstageinstanceid in ( select programstageinstanceid from programstageinstance where programstageid in ("
+        "select programstageid from programstage where programid in (select programid from program where uid in " + trackers + ")));" + "\n")
+
+    f.write("delete from trackedentityattributevalue where trackedentityinstanceid "
+          "in ( select trackedentityinstanceid from programinstance where programid in(select programid from program where uid in " + trackers + "));" + "\n")
+    f.write("delete from trackedentityattributevalueaudit where trackedentityinstanceid "
+          "in ( select trackedentityinstanceid from programinstance where programid in(select programid from program where uid in " + trackers + "));" + "\n")
+
+    f.write(
+        "delete from programstageinstance where programstageid in ("
+        "select programstageid from programstage where programid in (select programid from program where uid in " + trackers + "));" + "\n")
+
+    f.write("delete from programstageinstance where programstageinstanceid "
+          "in ( select psi.programstageinstanceid  from programstageinstance psi "
+          "inner join programstage ps on ps.programstageid=psi.programstageid "
+          "inner join program p on p.programid=ps.programid "
+          "where p.uid in " + trackers + ");" + "\n")
+
+    f.write(
+        "create view tei_to_remove as select trackedentityinstanceid \"teiid\" from programinstance where programid in (select programid from program where uid in " + trackers + ");" + "\n")
+    f.write(
+        "delete from programinstance where programid in (select programid from program where uid in " + trackers + ");" + "\n")
+    f.write("delete from programinstance where trackedentityinstanceid in ( select * from tei_to_remove);" + "\n")
+    f.write("delete from trackedentityinstance where trackedentityinstanceid in ( select * from tei_to_remove);" + "\n")
+    f.write("delete from trackedentityprogramowner where trackedentityinstanceid in ( select * from tei_to_remove);" + "\n")
+    f.write("drop view tei_to_remove ;" + "\n")
+
+    # f.write("delete from programstageinstancecomments where programstageinstanceid "
+    #           "in ( select psi.programstageinstanceid  from programstageinstance psi "
+    #           "inner join programstage ps on ps.programstageid=psi.programstageid "
+    #           "inner join program p on p.programid=ps.programid "
+    #           "where p.uid in " + trackers + ");")
+    f.write("--remove tracker finish" + "\n")
+
+
+def convert_to_sql_format(list_uid):
+    if len(list_uid) == 0:
+        return ""
+    return json.dumps(list_uid).replace("\"", "\'").replace("[", "(").replace("]", ")")
```

Comment:
That's a bit hacky. We can do the mapping with python constructs, example: `"(" + ", ".join(["'{}'".format(uid) for uid in list_uid]) + ")"`.